pinot-core/src/main/java/org/apache/pinot/core/query/reduce/AggregationDataTableReducer.java

```diff
@@ -102,6 +103,7 @@ private void reduceWithIntermediateResult(DataSchema dataSchema, Collection<Data
           intermediateResults[i] = _aggregationFunctions[i].merge(mergedIntermediateResult, intermediateResultToMerge);
         }
       }
+      Tracing.ThreadAccountantOps.sampleAndCheckInterruption();
```

Comment:
Why isn't this done at the beginning of the reduce call / method ?

Reply:
because at that point the memory usage would be fairly low. Actually placing this at the beginning vs end doesn't matter a lot as aggregation doesn't have that many loops